#print(3/0)
#当程序遇到问题时不让程序结束，而越过错误继续向下执行。
"""
try……except……else

格式：
    语句t
except错误码 as e：
    语句1
except错误码 as e：
    语句2
……

else：
    语句e

注意：else语句可有可无
作用：用来检测try语句块中的错误，从而让except语句补货错误信息
并处理

逻辑：当程序执行到try——except-else语句时
1、如果当try“语句t”执行出现错误，会匹配第一个错误码，
如果匹配上就执行对应的“语句”
2、如果当try“语句t”执行出现错误，没有匹配的异常，
错误将会被提交到上一层的try语句。或者到程序的最上层
3、如果当try“语句t”执行没有出现错误，执行else下的“语句e”

"""

try:
    print(3/1)
except ZeroDivisionError as e:
    print("除数为0")
except NameError as e:
    print("没有该变量")
else:
    print("代码没问题")


print("**********************")







#使用except而不是用任何错误类型
try:
    print(3/0)
    print(num)
except:
    print("程序出现异常")

#使用except带着多种异常
try:
    #pass
    print(num)
except(NameError,ZeroDivisionError):
    print("程序出现NameError或ZeroDivisionError")
    #视频51
